src/plugins/shipping-rates/mutations/createFlatRateFulfillmentRestriction.js

```diff
@@ -25,6 +25,9 @@ export default async function createFlatRateFulfillmentRestrictionMutation(conte
 
   await context.validatePermissions("reaction:shippingRestrictions", "create", { shopId, legacyRoles: ["owner", "admin", "shipping"] });
 
+  restriction._id = Random.id();
+  restriction.shopId = shopId;
+
   const { insertedCount } = await FlatRateFulfillmentRestrictions.insertOne({
     _id: Random.id(),
```

Comment:
Line 32 and 33 are overriding the values set on lines 28-29. Delete lines 32-33 and then just pass the restriction object without destructuring